### PR TITLE
[ART-3908] refactor check bugs

### PR DESCRIPTION
**Pull request by locriandev**

### CODE REVIEW

File: pyartcd/pyartcd/pipelines/check_bugs.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+        # Find bugs
+        cmd = [
+            ELLIOTT_BIN,
+            f'--group=openshift-{version}',
+            f'--working-dir={version}-working',
+            'find-bugs',
+            '--mode=sweep'
+        ]
+        self.logger.info(f'Executing command: {" ".join(cmd)}')
+        process = subprocess.Popen(cmd, stdout=subprocess.PIPE)
```
Comment (joepvd) [inline pyartcd/pyartcd/pipelines/check_bugs.py:157]: Having stderr when it fails would be great as well

File: pyartcd/pyartcd/pipelines/check_bugs.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
+import os
```
Comment (joepvd) [inline pyartcd/pyartcd/pipelines/check_bugs.py:1]: some tests would be great

File: pyartcd/pyartcd/pipelines/check_bugs.py [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+            'find-bugs',
+            '--mode=blocker',
+            '--report',
+            '--output=slack'
+        ]
+        self.logger.info(f'Executing command: {" ".join(cmd)}')
+
+        process = subprocess.Popen(cmd, stdout=subprocess.PIPE)
+        out, _ = process.communicate()
+        errcode = process.returncode
```
Comment (joepvd) [inline pyartcd/pyartcd/pipelines/check_bugs.py:124]: mark the job as failed if one or more things fail. then we know we have to look.
optional, a retry of a failed command to reduce flakes. but we can get to that if there would be too many flakes

**Review by joepvd — COMMENTED** [review]
Nice! Like this direction!

File: pyartcd/pyartcd/pipelines/check_bugs.py (hunk 2 above)
Comment (locriandev) [reply pyartcd/pyartcd/pipelines/check_bugs.py:1]: done here: https://github.com/openshift/aos-cd-jobs/pull/3003/commits/7e42210bf38259cb2f3bc265ac07f4855700a287

File: pyartcd/pyartcd/pipelines/check_bugs.py (hunk 3 above)
Comment (locriandev) [reply pyartcd/pyartcd/pipelines/check_bugs.py:124]: Done here: https://github.com/openshift/aos-cd-jobs/pull/3003/commits/281e546c8bd9475dcb2d19aa2d7f8cfbe46d0f1c
I'd leave retries for later, if we see we actually need this

File: pyartcd/pyartcd/pipelines/check_bugs.py (hunk 2 above)
Comment (locriandev) [reply pyartcd/pyartcd/pipelines/check_bugs.py:1]: Done here: https://github.com/openshift/aos-cd-jobs/pull/3003/commits/4f6bd28a0cd4a104b3807e5cb38bad4a26f87492

**Pull request merged**